src/modules/land_detector/FixedwingLandDetector.cpp

```diff
@@ -71,9 +71,18 @@ bool FixedwingLandDetector::_get_landed_state()
 
 	} else if (hrt_elapsed_time(&_vehicle_local_position.timestamp) < 1_s) {
 
-		// Horizontal velocity complimentary filter.
-		float val = 0.97f * _velocity_xy_filtered + 0.03f * sqrtf(_vehicle_local_position.vx * _vehicle_local_position.vx +
-				_vehicle_local_position.vy * _vehicle_local_position.vy);
+		float val;
+
+		if (!_vehicle_local_position.v_xy_valid) {
+			// set _velocity_xy_filtered to 0 if data is invalid
+			val = 0.0f;
```

Comment:
We may need to add some logic that makes the acceleration threshold tighter if we don't have ground speed and airspeed. Similar to what we already do for the ground speed check if there is no airspeed.
Otherwise it gets dangerous to fly planes without GPS and without airspeed sensor, as then they could detect "landing" in-air if flies very smoothly.

Reply:
I was thinking the same, but what I am struggling to understand is why this wasn't already an issue in the first half of the flight (before the filtered horizontal velocity diverges): 

![image](https://github.com/user-attachments/assets/0d280a78-5216-4bc4-9cc9-1646db7dd15d)